src/components/FormAlertWrapper.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'underscore';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {withNetwork} from './OnyxProvider';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import RenderHTML from './RenderHTML';
+import TextLink from './TextLink';
+import Text from './Text';
+import colors from '../styles/colors';
+import compose from '../libs/compose';
+import OfflineIndicator from './OfflineIndicator';
+import networkPropTypes from './networkPropTypes';
+import styles from '../styles/styles';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+
+const propTypes = {
+    /** Wrapped child components */
+    children: PropTypes.node.isRequired,
+
+    /** Styles for container element */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Whether to show the alert text */
+    isAlertVisible: PropTypes.bool.isRequired,
+
+    /** Whether message is in html format */
+    isMessageHtml: PropTypes.bool,
+
+    /** Error message to display above button */
+    message: PropTypes.string,
+
+    /** Props to detect online status */
+    network: networkPropTypes.isRequired,
+
+    /** Callback fired when the "fix the errors" link is pressed */
+    onFixTheErrorsLinkPressed: PropTypes.func,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    containerStyles: [],
+    isMessageHtml: false,
+    message: '',
+    onFixTheErrorsLinkPressed: () => {},
+};
+
+// The FormAlertWrapper offers a standardized way of showing error messages and offline functionality.
+//
+// This component takes other components as a child prop. It will then render any wrapped components as a function using "render props",
+// and passes it a (bool) isOffline parameter. Child components can then use that isOffline to determine offline behavior.
```

Comment:
It would be helpful to add a link to the react docs about render props (just to save people the step of googling it)
```suggestion
// and passes it a (bool) isOffline parameter. Child components can then use the isOffline variable to determine offline behavior.
```